src/components/GenericDropdown/GenericDropdown.tsx

```diff
@@ -0,0 +1,70 @@
+import { observer } from 'mobx-react';
+import * as React from 'react';
+
+import { DropdownProps, DropdownOption } from '../../common-elements/dropdown';
+import { DropdownLabel, DropdownWrapper } from '../PayloadSamples/styled.elements';
+
+export interface GenericDropdownProps<T> {
+  items?: T[];
+  options: DropdownOption[];
+  label?: string;
+  renderDropdown: (props: DropdownProps) => JSX.Element;
+  children: (activeItem: T) => JSX.Element;
+}
+
+export interface GenericDropdownState {
+  activeItemIdx: number;
+}
+
+@observer
+export class GenericDropdown<T> extends React.Component<
```

Comment:
I would rename this component to something like `GenericChildrenSwitcher` or maybe you have other ideas. `GenericDropdown` is not what this component is as it accept an option `renderDropdown` which is a bit confusing.

Reply:
What about moving the logic of rendering dropdown/label into this GenerciDropdown component? And just passing Dropdown/Label as props?